formatter.go

```diff
@@ -14,33 +15,62 @@ import (
 // the write entry
 type Formatter interface {
 	Format(w io.Writer, logLevel LogLevel, msg string, timestamp time.Time, fields Fields) error
+	AddMapping(fieldMapping FieldMapping)
 }
 
 // JSONFormatter used to output logs in JSON format
 type JSONFormatter struct {
-	Compact bool
+	FieldMapping FieldMapping
+	Compact      bool
 }
 
 func (j JSONFormatter) getKey(key string) string {
 	if j.Compact {
-		return "@" + key[:1]
+		var mappedKey string
+		var ok bool
+
+		if mappedKey, ok = j.FieldMapping[key]; !ok {
+			mappedKey = key
+		}
+
+		return mappedKey
 	}
 	return key
 }
 
+func (j *JSONFormatter) AddMapping(fieldMapping FieldMapping) {
```

Comment:
The naming of the method is confusing. AddMapping suggests that you add items to an existing FieldMapping instance. But instead it's overwritten. Second, AddMapping is too generic. A suggestion is to rename to SetFieldMappings. Alternative if you really want to append to exising FieldMapping entries, rename to AddFieldMappings and adjust the code accordingly

Reply:
Renamed it to `SetFieldMapping`